### DISCUSSION TITLE
How can we deactivate the Gaze Interactor in the presence of hands or controllers within the scene?

### DISCUSSION THREAD

**Discussion by Ram1508** [MixedRealityToolkit/MixedRealityToolkit-Unity#468] [Q&A]
@AMollis

[Scenario]
I'm faced with a situation where I want to dynamically suppress the gaze interactor whenever hand interactors are present on the screen.

[Approach]

In a scene featuring two buttons, "Enable Gaze" and "Enable Hand," I've employed the following methods:

1) When the "Enable Hand" button is activated:

I've configured the **Interaction Layer Mask** to be set to **Nothing** in the FuzzyGazeInteractor script (Gaze Interactor) and GazePinchInteractor Script (Right and Left Hand Controller).
 
![Gaze-Nothing](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/131968892/7180b42f-586e-417b-946f-01f68ac9ed6d)

2) When "Enable Gaze" button is activated, I've set the **Interaction Layer Mask** to be **Everything.**

![GAze-Everything](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/131968892/3b3175b7-8eb1-4a31-8dfb-c483c28e9ea1)

[Question]
Is this approach the most effective way to achieve this, or are there alternative methods that might be more suitable?

**Comment by marlenaklein-msft** [comment]
Hi @Ram1508, curious is there a reason you are modifying the InteractionLayerMask instead of enabling/disabling the GazePinchInteractor? I believe the DisableInteractorsExample scene and the InteractorBehaviorControls sample script have similar examples if that's helpful

**Comment by Ram1508** [comment]
@marlenaklein-msft 

Thanks for the reply.

No, there is no specific reason to modifying the interaction layer mask. As I was exploring for different possibilities, I encountered with that method. I tried  the DisableInteractorsExample scene and it was helpful.
